### PR TITLE
Updates scikit-learn version to 0.22

### PR DESCRIPTION
**Pull request by gui-miotto**

### CODE REVIEW

File: test/test_pipeline/components/feature_preprocessing/test_extra_trees_regression.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def test_default_configuration_regression(self):
             predictor = regressor.fit(X_train_trans, Y_train)
             predictions = predictor.predict(X_test_trans)
             accuracy = sklearn.metrics.mean_squared_error(predictions, Y_test)
-            self.assertAlmostEqual(accuracy, 20.193400000000004, places=2)
+            self.assertAlmostEqual(accuracy, 17.93480710955056, places=2)
```
Comment (gui-miotto) [inline test/test_pipeline/components/feature_preprocessing/test_extra_trees_regression.py]: The accuracies decreased. Couldn't figure out why, even after carefully inspecting the list of changes.

Comment (mfeurer) [reply test/test_pipeline/components/feature_preprocessing/test_extra_trees_regression.py]: The metric is actually misnamed. Could you please change the metric to `error`? Then it's clear that it actually improved by going to 0.22.2.

File: test/test_pipeline/components/regression/test_ridge_regression.py [hunk 2]
```diff
@@ -11,11 +11,12 @@ class RidgeComponentTest(BaseRegressionComponentTest):
...
     res["default_boston"] = 0.7035465377559671
     res["default_boston_iterative"] = None
-    res["default_boston_sparse"] = 0.1086549402189615
+    res["default_boston_sparse"] = 0.1163004000785135
+    res["default_boston_sparse_places"] = 5
     res["default_boston_iterative_sparse"] = None
     res["default_diabetes"] = 0.32614416980439365
     res["default_diabetes_iterative"] = None
-    res["default_diabetes_sparse"] = 0.1298841767742679
+    res["default_diabetes_sparse"] = 0.12989753681434824
```
Comment (gui-miotto) [inline test/test_pipeline/components/regression/test_ridge_regression.py]: Apparently ridge regression had a small bug when dealing with X sparse. It has been fixed in v0.22.
Model details here: https://github.com/scikit-learn/scikit-learn/pull/13995

File: test/test_pipeline/components/feature_preprocessing/test_extra_trees_regression.py (hunk 1 above)
Comment (gui-miotto) [reply test/test_pipeline/components/feature_preprocessing/test_extra_trees_regression.py]: Ah! This is good news. :-)
I'll rename it.

File: test/test_evaluation/test_train_evaluator.py [hunk 3]
```diff
@@ -393,7 +393,7 @@ def test_partial_cv(self, pipeline_mock):
         self.assertRaises(queue.Empty, evaluator.queue.get, timeout=1)
 
         self.assertEqual(evaluator.file_output.call_count, 0)
-        self.assertEqual(rval['loss'], 0.46666666666666667)
+        self.assertEqual(rval['loss'], 0.5)
```
Comment (mfeurer) [inline test/test_evaluation/test_train_evaluator.py]: Do you have any idea what happened to these few test for cross-validation?

Comment (gui-miotto) [reply test/test_evaluation/test_train_evaluator.py]: TL;DR: @mfeurer , you were right, sklearn 21.3 and 22.1 generate different splits for the same given seed.

More details (level 1): Not only the chosen samples differ, but also the number of samples. For instance, this line that you highlighted: both the development branch and this PR's branch get 8 itens of the test set correctly classified. However, the split generated by sklearn 21.3 has 15 samples, an the one generated by 22.1 has 16, therefore the losses are respectively 0.46667 and 0.5.

More details (level 2): In order to double check this hypothesis, I performed one more sanity check: 
* run the test using the development branch and sklearn 22.1 and
* run the test using this PR's branch with sklearn 21.3. 
(In other words, I tried the old sklearn with the new branch and the new sklearn with the old branch.)
The accuracy I got for the 1st experiment was .46667 and for the 2nd it was .5. This supports the hypothesis that the cause of this discrepancy has nothing to do with auto-sklearn's code.

Comment (mfeurer) [reply test/test_evaluation/test_train_evaluator.py]: Okay, thanks a lot for checking this.
